hls-plugin-api/src/Ide/Types.hs

```diff
@@ -277,23 +281,52 @@ data PluginDescriptor (ideState :: Type) =
                    , pluginNotificationHandlers :: PluginNotificationHandlers ideState
                    , pluginModifyDynflags :: DynFlagsModifications
                    , pluginCli            :: Maybe (ParserInfo (IdeCommand ideState))
-                   , pluginFileType       :: [T.Text]
+                   , pluginFileType       :: PluginFileType
                    -- ^ File extension of the files the plugin is responsible for.
                    --   The plugin is only allowed to handle files with these extensions.
                    --   When writing handlers, etc. for this plugin it can be assumed that all handled files are of this type.
                    --   The file extension must have a leading '.'.
                    }
 
+-- A description of the types of files that the plugin
```

Comment:
```suggestion
-- | A description of the types of files that the plugin
```
I realise I'm just repeating myself, but again I don't see why we can't let plugins just fail on dependency files if they need stuff that can't be provided.